### PR TITLE
Feature/cx 1706 unbundle dependencies for npm

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem

* As a customer/developer I don't want to have repeated dependencies in my final bundle as means to economize space.
* As a developer I want to run npm audit and actually be able to fix vulnerabilities in the SDK

# Solution
* Make sure dependencies are not in the final bundle
* Do no lock patch versions

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [ ] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have tests passed locally?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

File: dist/onfido.min.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
...xports=o},function(e,t,n){e.exports={"absolute-center":"onfido-sdk-ui-Theme-absolute-center",step:"onfido-sdk-ui-Theme-step",fullScreenStep:"onfido-sdk-ui-Theme-fullScreenStep",fullHeightContainer:"onfido-sdk-ui-Theme-fullHeightContainer",fullHeightMobileContainer:"onfido-sdk-ui-Theme-fullHeightMobileContainer",navigationBar:"onfido-sdk-ui-Theme-navigationBar",content:"onfido-sdk-ui-Theme-content",scrollableContent:"onfido-sdk-ui-Theme-scrollableContent",fullScreenContentWrapper:"onfido-sdk-ui-Theme-fullScreenContentWrapper",footer:"onfido-sdk-ui-Theme-footer",thickWrapper:"onfido-sdk-ui-Theme-thickWrapper","mbottom-large":"onfido-sdk-ui-Theme-mbottom-large",center:"onfido-sdk-ui-Theme-center",icon:"onfido-sdk-ui-Theme-icon",help:"onfido-sdk-ui-Theme-help",header:"onfido-sdk-ui-Theme-header",helpList:"onfido-sdk-ui-Theme-helpList"}},function(e,t,n){var r=n(165),o=n(171),i=n(175),a=n(179),s=n(141),c=n(142),u=n(1);e.exports=function(e){for(var t=1;t<arguments.length;t++){var n,l=null!=arguments[t]?arguments[t]:{},p=c(l);"function"==typeof s&&(p=a(p).call(p,i(n=s(l)).call(n,function(e){return o(l,e).enumerable}))),r(p).call(p,function(t){u(e,t,l[t])})}return e}},function(e,t,n){var r;
```
Comment (rfreitas) [inline dist/onfido.min.js]: we should only update the dist for release, if possible it would be good to remove this diff

File: package.json [hunk 2]
```diff
@@ -105,33 +105,36 @@
...
-    "reselect": "2.5.1",
+    "redux": "~3.5.2",
+    "reselect": "~2.5.1",
     "socket.io-client": "^2.0.4",
-    "supports-webp": "1.0.3",
-    "visibilityjs": "1.2.4",
-    "wpt": "https://github.com/Woopra/js-client-tracker/archive/v5.1.19.tar.gz"
+    "supports-webp": "~1.0.3",
+    "visibilityjs": "~1.2.4",
+    "react": "^15.0.0",
```
Comment (rfreitas) [inline package.json]: I don't think we call this directly, right? If so, I don't think it needs to go on the dependency lit

File: dist/onfido.min.js (hunk 1 above)
Comment (kopijunkie) [reply dist/onfido.min.js]: Have rolled back the changed minified file so it matches what's on `development` now

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

File: package.json (hunk 2 above)
Comment (rfreitas) [reply package.json]: for summary: this impacts the npm artifact in 0.2 MB. this is already a big artifact, so we will proceed until we consolidate react and preact

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-698-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://698-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**